ObjectMemoryTests.c

```diff
@@ -435,16 +435,37 @@ Test(DiscoverByteLengthOfObjectWithOddSize) {
   Expect(byteLength == (size - 2)*2 - 1);
 }
 
-Test(AllocateSmallObject) {
-  ObjectPointer objectPointer, classPointer = 0x2468, reportedClass;
+Test(AllocateSmallObjectByReclaimingRightSizedSpace) {
+  ObjectPointer objectPointer, anObject = 0x2300, classPointer = 0x2468, reportedClass, location = 0x3200;
   Word size, wordLength = 10;
   Bool pointerBit;
-  /* this test actually allocates an object, so let's ensure there's a free space */
+  /* this test allocates an object, so let's ensure there's a free space exactly equal to the object size */
   Word segment = (1 % HeapSegmentCount);
   currentSegment = segment;
   /* add a valid pointer to the free chunk list */
-  ObjectMemory_headOfFreeChunkList_inSegment_put(wordLength + HeaderSize, currentSegment, 0x2300);
+  ObjectMemory_locationBitsOf_put(anObject, location);
+  ObjectMemory_segmentBitsOf_put(anObject, segment);
+  ObjectMemory_headOfFreeChunkList_inSegment_put(wordLength + HeaderSize, currentSegment, anObject);
+
+  objectPointer = ObjectMemory_instantiateClass_withPointers(classPointer, wordLength);
+
+  Expect(objectPointer != NilPointer);
+
+  size = ObjectMemory_sizeBitsOf(objectPointer);
+  Expect(size == wordLength + HeaderSize);
 
+  reportedClass = ObjectMemory_fetchClassOf(objectPointer);
+  Expect(reportedClass == classPointer);
+
+  pointerBit = ObjectMemory_pointerBitOf(objectPointer);
+  Expect(pointerBit == YES);
+}
+
+Test(AllocateSmallObjectBySplittingLargeChunk) {
```

Comment:
This test is new! It relies on the initial, clean state of the `ObjectMemory` and checks that it is possible to allocate an object from that position (i.e. that we can find a too-big chunk on the `BigSize` chunk list, fragment it, and use the fragment to fulfil the allocation).